packages/server/lib/clients/email.client.ts

```diff
@@ -1,43 +1,92 @@
 import formData from 'form-data';
 import Mailgun from 'mailgun.js';
-
+import type { Transporter } from 'nodemailer';
+import nodemailer from 'nodemailer';
 import { getLogger } from '@nangohq/utils';
-
-import type { Interfaces } from 'mailgun.js/definitions';
+import type { Interfaces, MessagesSendResult } from 'mailgun.js/definitions';
+import { envs } from '../env.js';
 
 const logger = getLogger('Server.EmailClient');
 
+interface EmailProvider<T> {
+    send(email: string, subject: string, html: string): Promise<T>;
+}
+
 export class EmailClient {
     private static instance: EmailClient | undefined;
-    private client: Interfaces.IMailgunClient;
+    private provider: MailgunEmailProvider | SmtpEmailProvider | NoEmailProvider; // Mailgun specific provider is here for legacy reason
 
-    private constructor(config: { username: string; key: string; url?: string }) {
-        const mailgun = new Mailgun(formData);
-        this.client = mailgun.client(config);
+    private constructor() {
+        if (envs.MAILGUN_API_KEY) {
+            this.provider = new MailgunEmailProvider();
+        } else if (envs.SMTP_HOST) {
+            this.provider = new SmtpEmailProvider();
+        } else {
+            this.provider = new NoEmailProvider();
+        }
     }
 
     static getInstance() {
         if (!EmailClient.instance) {
-            EmailClient.instance = new EmailClient({
-                username: 'api',
-                key: process.env['MAILGUN_API_KEY'] || 'EMPTY',
-                url: process.env['MAILGUN_URL'] || ''
-            });
+            EmailClient.instance = new EmailClient();
         }
         return EmailClient.instance;
     }
 
     send(email: string, subject: string, html: string) {
-        if (process.env['MAILGUN_API_KEY'] === undefined || process.env['MAILGUN_API_KEY'] === 'EMPTY' || !this.client) {
-            logger.info('Email client not configured');
-            logger.info('The following email would have been sent:');
-            logger.info(email, subject);
-            logger.info(html);
-            return;
-        }
+        return this.provider.send(email, subject, html);
+    }
+}
+
+class NoEmailProvider implements EmailProvider<void> {
+    // eslint-disable-next-line @typescript-eslint/require-await
+    async send(email: string, subject: string, html: string): Promise<void> {
+        logger.info('Email client not configured');
+        logger.info('The following email would have been sent:');
+        logger.info(email, subject);
+        logger.info(html);
+    }
+}
+
+class SmtpEmailProvider implements EmailProvider<void> {
+    private transporter: Transporter;
+
+    constructor() {
+        this.transporter = nodemailer.createTransport({
+            host: envs.SMTP_HOST,
+            port: envs.SMTP_PORT,
+            secure: envs.SMTP_SECURE,
+            auth: {
+                user: envs.SMTP_USER,
+                pass: envs.SMTP_PASSWORD
```

Comment:
There is no way to pass all of those envs in a single URL?

Reply:
good question. let me check if nodemailer support a url, I don't really want to parse the url ourselves